### PR TITLE
Add 'domain' and 'smtp from' config entries

### LINKED ISSUE #193: Bug: Configure reset password mail sender address
https://github.com/rr-/szurubooru/blob/1fe22a4d0aa6f7e374394afc7777361e48f21823/server/szurubooru/api/password_reset_api.py#L32

Using name as the sender is not really suitable because `name` is usually not a valid domain name. Also, if we can add a config to allow user to set it to match the DNS records, these will reduce the chance of the mail going to spam.

### PR DESCRIPTION
**Pull request by neobooru**
Fixes #193 and #256

This however requires users to manually set the domain and no-reply address in the config.yaml.
The `domain` field currently is optional, but it would probably be better to make it required and not fall back to HTTP_ORIGIN and HTTP_REFERER, which might be inaccurate or not set (especially behind reverse proxies and the like). 

The `smtp:from` field is required, the server will crash without it set.

This also should break any docker configs that currently exist (though I haven't looked at those).

### CODE REVIEW

File: server/config.yaml.dist [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 # shown in the website title and on the front page
 name: szurubooru
+# full url to the homepage of this szurubooru site, with no trailing slash
+domain: http://example.com
```
Comment (sgsunder) [inline server/config.yaml.dist]:
```suggestion
domain: # example: http://example.com
```

Probably better to leave it blank by default

**Comment by sgsunder** [comment]
Sorry it took me so long to get to this.

I'm not sure what you mean by:
> The `smtp:from` field is required

I'm guessing you only mean it's required if you have email-based password resets enabled (i.e. `smtp:host` is set), correct? All the SMTP fields should remain optional if the user only opts for manual password resets.

For the users who do use email-based resets, perhaps we can generate a more friendly and descriptive exception letting users know that `smtp:from` is a new, added parameter that should be set if they haven't already.

**Comment by neobooru** [comment]
Seems fair. Is there already a certain way to throw exceptions in this codebase, or should I just do a descriptive `raise Exception(msg)` when `smtp:from` isn't set? And should that throw on startup or when the user actually tries to send a mail (on startup would probably be better).

**Comment by sgsunder** [comment]
You should probably add the check to `validate_config()` in `facade.py` (which is the application entrypoint, so it would be at startup). Seems like there's already a `errors.ConfigError` defined which you can throw.

If you want to display warnings about stuff like `domain` and such, you could probably call `logging.warning()` within the same function.

**Comment by sgsunder** [comment]
Merged manually with 80d272d , thanks!

**Pull request closed without merging**